i18n/validate.py

```diff
@@ -37,7 +37,7 @@ def validate_po_files(configuration, locale_dir, root_dir=None, report_empty=Fal
 
             # Validate only .po files that are not product of a merge (see generate.py).
             # If django-partial.po has a problem, then django.po will also, so don't report it.
-            if ext.lower() == '.po' and os.path.basename(filename) not in merged_files:
+            if ext.lower() == '.po' and (os.path.basename(filename) not in merged_files or check_all):
```

Comment:
Nit: I think it's slightly easier to read when `check_all` comes before `os.path.basename(filename) not in merged_files`.

```
if ext.lower() == '.po' and (check_all or os.path.basename(filename) not in merged_files):
```

Reply:
It would probably be good to update the comment that precedes this.